File: app/src/main/java/com/sofps/inspirationalquotes/data/QuotesService.java
package com.sofps.inspirationalquotes.data;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface QuotesService {

    @GET("qod.json")
    Call<QuoteApi> getQuoteOfTheDay();

    @GET("qod.json")
    Call<QuoteApi> getQuoteOfTheDayForCategory(@Query("category") String category);

    @GET("quote/random.json")
    Call<QuoteApi> getRandomQuote();
}

File: app/src/main/java/com/sofps/inspirationalquotes/data/QuotesService.kt
package com.sofps.inspirationalquotes.data

import retrofit2.http.GET
import retrofit2.http.Query

interface QuotesService {

    @GET("qod.json")
    suspend fun getQuoteOfTheDay(): QuoteApi

    @GET("quote/random.json")
    suspend fun getRandomQuote(): QuoteApi

    @GET("qod.json")
    suspend fun getQuoteOfTheDayForCategory(@Query("category") category: String): QuoteApi
}
